internal/services/springcloud/spring_cloud_dynatrace_application_performance_monitoring_resource.go

```diff
@@ -0,0 +1,344 @@
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"spring_cloud_service_id": commonschema.ResourceIDReferenceRequiredForceNew(appplatform.SpringId{}),
+
+		"globally_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"connection_point": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"environment_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"tenant_token": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"api_url": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"api_token": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"tenant": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (s SpringCloudDynatraceApplicationPerformanceMonitoringResource) Attributes() map[string]*schema.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (s SpringCloudDynatraceApplicationPerformanceMonitoringResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model SpringCloudDynatraceApplicationPerformanceMonitoringModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.AppPlatform.AppPlatformClient
+			springId, err := appplatform.ParseSpringID(model.SpringCloudServiceId)
+			if err != nil {
+				return fmt.Errorf("parsing spring service ID: %+v", err)
+			}
+			id := appplatform.NewApmID(springId.SubscriptionId, springId.ResourceGroupName, springId.SpringName, model.Name)
+
+			existing, err := client.ApmsGet(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(s.ResourceType(), id)
+			}
+
+			resource := appplatform.ApmResource{
+				Properties: &appplatform.ApmProperties{
+					Type: "Dynatrace",
+					Properties: utils.ToPtr(map[string]string{
+						"api-url":          model.ApiUrl,
+						"connection_point": model.ConnectionPoint,
+						"environment-id":   model.EnvironmentId,
+					}),
+					Secrets: utils.ToPtr(map[string]string{
+						"api-token":   model.ApiToken,
+						"tenanttoken": model.TenantToken,
+						"tenant":      model.Tenant,
+					}),
+				},
+			}
+			err = client.ApmsCreateOrUpdateThenPoll(ctx, id, resource)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			if model.GloballyEnabled {
```

Comment:
Shall we rename this property to `global_enabled`?

Reply:
I think it's better to keep it, because it's aligned with the API and also with a similar resource: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/springcloud/spring_cloud_application_insights_application_performance_monitoring_resource.go#L62